src/RenderMethods/RenderFieldTemplate.tsx

```diff
@@ -0,0 +1,132 @@
+import type { FieldTemplateProps } from '@rjsf/core';
+import { useVerna } from '../VernaContextProvider';
+import { getCurrentField, getCurrentSection, makeid } from '../utils';
+import { stringDefinition, sectionDefinition } from '../schemaComponents/templates';
+import _ from 'lodash';
+import { JSONSchema7 } from 'json-schema';
+
+// This component is used to render every widget
+// Its purpose here is to add functionality common to every of those such as add or remove
+// It will be used to add drag and drop area later
+
+export default function RenderFieldTemplate({ id, schema, children }: FieldTemplateProps) {
+  const { schema: vernaSchema, setSchema, uiSchema, setUiSchema } = useVerna();
+  const canAddField = id.split('_').length === 3;
+  const canAddSection = id.split('_').length === 2;
+  const isRoot = id === 'root';
+  const ownProperties = Object.keys(schema.properties || {}).length === 0;
+  const currentSection = getCurrentSection(id);
+  const currentField = getCurrentField(id);
+
+  function addItemToSchema(newKey: string) {
+    const newSchema = { ...vernaSchema };
+    if (newSchema.properties) {
+      const section = newSchema.properties[currentSection] as JSONSchema7;
+      if (section.properties) {
+        section.properties[newKey] = stringDefinition(newKey);
+      }
+    }
+    setSchema(newSchema);
+  }
+
+  function addItemToUiSchemaOrder(newKey: string) {
+    const newUiSchema = { ...uiSchema };
+    if (currentSection in newUiSchema && 'ui:order' in newUiSchema[currentSection]) {
+      const currentSectionIndex = _.findIndex(
+        uiSchema[currentSection]['ui:order'],
+        (key) => key === currentField,
+      );
```

Comment:
There is a built-in for that.
```suggestion
      const currentSectionIndex =  uiSchema[currentSection]['ui:order'].findIndex(
        (key) => key === currentField,
      );
```